components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/endpoint/registry/impl/EndpointRegistryImpl.java

```diff
@@ -251,7 +251,8 @@ public void updateEndpointRegistry(String registryId, String registryName, Endpo
             tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
                     .getTenantId(tenantDomain);
             // if another registry with the updated name or type already exists, fail the operation.
-            if (registryDAO.isEndpointRegistryTypeExists(endpointRegistryInfo.getType(), tenantId)) {
+            if (!registryType.equals(endpointRegistryInfo.getType()) &&
+                    registryDAO.isEndpointRegistryTypeExists(endpointRegistryInfo.getType(), tenantId)) {
                 EndpointRegistryUtil.handleResourceAlreadyExistsException("Endpoint Registry of type '"
```

Comment:
Since this is throwing an Exception use raiseException instead of handleException

Reply:
rename the handleResourceAlreadyExistsException method ?

Reply:
Done